.github/workflows/release.yml

```diff
@@ -1,32 +1,61 @@
-# This is a basic workflow to help you get started with Actions
-
 name: Release
 
-# Controls when the action will run. Triggers the workflow on push or pull request
-# events but only for the master branch
 on:
   push:
     tags:
-      - v*.*.*
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+
+permissions:
+  contents: write
+
+env:
+  KUBEBUILDER_VERSION: 3.3.0
 
 jobs:
-  release:
+  go-version:
     runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.go-version.outputs.version }}
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Go
-      uses: actions/setup-go@v1
-      with:
-        go-version: 1.16.9
-    - name: Install kubebuilder
-      run: |
-        curl -L https://github.com/kubernetes-sigs/kubebuilder/releases/download/v2.3.1/kubebuilder_2.3.1_linux_amd64.tar.gz -o /tmp/kb.tgz
-        tar zxf /tmp/kb.tgz -C /tmp/
-    - name: Run GoReleaser
-      uses: goreleaser/goreleaser-action@v2.8.0
-      with:
-        args: release --skip-sign
-      env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        KUBEBUILDER_ASSETS: /tmp/kubebuilder_2.3.1_linux_amd64/bin/
-
+      - uses: actions/checkout@v2
+      - id: go-version
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+  release-github:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "${{ needs.go-version.outputs.version }}"
+      - name: Install kubebuilder
+        run: |
+          curl -L https://github.com/kubernetes-sigs/kubebuilder/releases/download/v${{ env.KUBEBUILDER_VERSION }}/kubebuilder_${{ env.KUBEBUILDER_VERSION }}_linux_amd64.tar.gz -o /tmp/kb.tgz
+          tar zxf /tmp/kb.tgz -C /tmp/
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v3.0.0
+        if: startsWith(github.ref, 'refs/tags/')
```

Comment:
Not sure we need this if the trigger is `on: push: tags`? Nothing else will trigger this workflow

Reply:
Yeah, just added it as a second safety net. It's not strictly necessary.
We can also drop it.